agent-lib/src/main/java/org/terracotta/angela/agent/AgentController.java

```diff
@@ -753,37 +749,32 @@ public Dirs(File kitDir, File workingDir, RemoteKitManager kitManager) {
     }
 
     static Optional<Dirs> discover(InstanceId instanceId, String hostName, Distribution distribution, License license, String kitInstallationName, String kitInstallationPath) {
+      File kitPath = null;
       if (kitInstallationPath == null) {
         RemoteKitManager kitManager = new RemoteKitManager(instanceId, distribution, kitInstallationName);
         if (!kitManager.isKitAvailable()) {
           return Optional.empty();
         }
-
+        kitPath = kitManager.installKit(license, Collections.singletonList(hostName));
         logger.info("Installing kit on host {} from {}", hostName, distribution);
-        return Optional.of(new Dirs(
-            kitManager.installKit(license, Collections.singletonList(hostName)),
-            kitManager.getWorkingDir().toFile(),
-            kitManager
-        ));
-
       } else {
         // DO NOT ALTER THE KIT CONTENT IF kitInstallationPath IS USED
-        File kitDir = new File(kitInstallationPath);
-        Path workingPath = Agent.WORK_DIR.resolve(instanceId.toString());
-        try {
-          Files.createDirectories(workingPath);
-          if (license != null) {
-            license.writeToFile(workingPath.toFile());
-          }
-        } catch (IOException e) {
-          logger.debug("Can not create {}", workingPath, e);
+        kitPath = new File(kitInstallationPath);
+      }
+      Path workingPath = Agent.WORK_DIR.resolve(instanceId.toString());
```

Comment:
we can remove the work dir from Dirs class now because it is always the same value locally and remotely

Reply:
i'm not sure I understand.  I think it is still useful as it defines a fixed point under user.dir

Reply:
My comment was because originally I did this `Dirs` class to keep a ref of the 3 different directories for local/remote. 

But now the work dir is the same (not same value but same code). So I was suggesting that if you want you could remote it from Dirs because there is no value anymore since it does not differ.

But we could keep it and have this `Dirs` class represent all the dirs for an "install" too.